indexbuilder.go

```diff
@@ -258,14 +269,25 @@ type Document struct {
 	SkipReason string
 
 	// Document sections for symbols. Offsets should use bytes.
-	Symbols []DocumentSection
+	Symbols         []DocumentSection
+	SymbolsMetaData []*Symbol
 }
 
-type docSectionSlice []DocumentSection
+type symbolSlice struct {
+	symbols  []DocumentSection
+	metaData []*Symbol
+}
 
-func (m docSectionSlice) Len() int           { return len(m) }
-func (m docSectionSlice) Swap(i, j int)      { m[i], m[j] = m[j], m[i] }
-func (m docSectionSlice) Less(i, j int) bool { return m[i].Start < m[j].Start }
+func (s symbolSlice) Len() int { return len(s.symbols) }
+
+func (s symbolSlice) Swap(i, j int) {
+	s.symbols[i], s.symbols[j] = s.symbols[j], s.symbols[i]
+	s.metaData[i], s.metaData[j] = s.metaData[j], s.metaData[i]
+}
+
+func (s symbolSlice) Less(i, j int) bool {
+	return s.symbols[i].Start < s.symbols[j].Start
```

Comment:
I know this applies to the older code as well, but is it possible to have two symbols with the same Start but different End?

Reply:
I haven't considered this possibility of ctags generating this overlapping metadata. I wonder this is a possible case with the native ctags generators.